hunts/models.py

```diff
@@ -47,27 +48,80 @@ def get_num_unlocked(self):
     def get_num_metas_solved(self):
         return self.puzzles.filter(Q(status=Puzzle.SOLVED), Q(is_meta=True)).count()
 
-    def get_solves_per_hour(self):
-        current_time = timezone.now()
+    def get_num_metas_unsolved(self):
+        return self.puzzles.filter(~Q(status=Puzzle.SOLVED), Q(is_meta=True)).count()
+
+    # Gets the number of puzzles that are either solved or feed into a solved meta.
+    def get_progression(self):
+        query = """
+        WITH RECURSIVE progression_puzzles (id) AS (
+            SELECT id
+            FROM puzzles_puzzle
+            WHERE (status = 'SOLVED' AND hunt_id = %s)
+            UNION
+            SELECT Metas.from_puzzle_id
+            FROM puzzles_puzzle_metas Metas
+            INNER JOIN progression_puzzles ON progression_puzzles.id = Metas.to_puzzle_id
+        )
+        SELECT id FROM progression_puzzles
+        """
+        progression_ids = Hunt.objects.raw(query, [str(self.pk)])
+        return len(list(progression_ids))
```

Comment:
Can we have the query select the whole puzzle row and then return the RawQuerySet of Puzzle objects? So SELECT * and then returning the Hunt.objects.raw result. Then the stats caller can call len(list(result)). I think this way we can reuse this function (which is computing a pretty interesting set of puzzles) for other things

Reply:
Lol I see @asdfryan and I both suggested different things. I wanted to reuse this for #322 or something along those lines

Reply:
I'll implement this and use len(list(result)) when passing into the page.